### PR TITLE
docs: Update about dependency handling of Helm Charts w/signoff

### LINKED ISSUE #10979: [Docs] Infrastructure Lifecycle: Manner of dependency deployment
### Current State
This page https://docs.meshery.io/guides/infrastructure-management/overview should more fully describe the dependency deployment process.

### Desired State
Please examine Meshery Server's golang surrounding the “pattern engine” to determine whether such dependencies are deployed using the Helm go client (as a helm chart) or using the k8s go client (as a manifest)?

---

### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)
- 📚 [Instructions for contributing to documentation](https://docs.meshery.io/project/contributing/contributing-docs)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/contributing/build-and-release)
- 🎨 Wireframes and [designs for Meshery UI](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI) in Figma [(open invite)](https://www.figma.com/team_invite/redeem/qJy1c95qirjgWQODApilR9)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](http://discuss.meshery.io) and [Community Slack](https://slack.meshery.io)

### PR DESCRIPTION
**Pull request by Jougan-0**
**Notes for Reviewers**

This PR fixes #10979 

- [x] Auto deployment of CRDs/controllers update 
- [x] Dependency Detection and Resolution

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by l5io** [comment]
🚀 Preview for commit 5a4cb808e8f0a716f578239ef304549c24def5db at: https://66494bd711b52549436c1118--meshery-docs-preview.netlify.app

**Comment by Jougan-0** [comment]
[Direct Link](https://66494bd711b52549436c1118--meshery-docs-preview.netlify.app/guides/infrastructure-management/overview#4-dependency-detection-and-resolution) to preview

File: docs/pages/guides/infrastructure-management/infrastructure-management.md [hunk 1]
```diff
@@ -43,14 +43,13 @@ See [Importing Applications]({{site.baseurl}}guides/configuration-management/imp
...
 </details>
 
 <h3>Auto-Deployment of CRDs and Operators</h3>
-
-Meshery automates the deployment of Custom Resource Definitions (CRDs) and operators based on the source from which a particular component was registered. Currently, by default Meshery automatically deploys components that are sourced from Artifact Hub (utilizing Helm Charts). (_Support for OCI registries is expected in the near future._)
+Meshery automates the deployment of Custom Resource Definitions (CRDs) and operators based on the source from which a particular component was registered. When a deployment request is made for a HelmChart without the upgrade flag, Meshery does not check for the existence of the HelmChart and always tries to install it. In cases where the chart/release already exists, the installation fails (from the Helm side), and Meshery errors out. Currently, by default, Meshery automatically deploys components that are sourced from Artifact Hub (utilizing Helm Charts). (_Support for OCI registries is expected in the near future._)
```
Comment (leecalcote) [inline docs/pages/guides/infrastructure-management/infrastructure-management.md:52]: An explanation about default behavior and how to control it is missing. @MUzairS15

**Review by leecalcote — CHANGES_REQUESTED** [review]

File: docs/pages/guides/infrastructure-management/infrastructure-management.md (hunk 1 above)
Comment (leecalcote) [reply docs/pages/guides/infrastructure-management/infrastructure-management.md:52]: @MUzairS15, also, when will the deployment modal have a checkbox for installing necessary dependencies?

Comment (MUzairS15) [reply docs/pages/guides/infrastructure-management/infrastructure-management.md:52]: The default behaviour is to try install w/o checking for existence of a release.

Currently, the control is missing. I will add it along with exposing the checkbox for installing dependencies (taking it now).

Comment (MUzairS15) [reply docs/pages/guides/infrastructure-management/infrastructure-management.md:52]: @Jougan-0  will you update the PR to include the above behaviour described in the comment? I have a open PR for the implementation

Comment (Jougan-0) [reply docs/pages/guides/infrastructure-management/infrastructure-management.md:52]: Yah, sure I need to add ```The default behaviour is to try install w/o checking for existence of a release.
I will add it along with exposing the checkbox for installing dependencies 
```
explaining this part may you please reference me the pr link.

Comment (leecalcote) [reply docs/pages/guides/infrastructure-management/infrastructure-management.md:52]: P.S. We use English here.

**Comment by l5io** [comment]
🚀 Preview for commit 9e3c3861a50c3634eaf81a298a1c9fdd445d9ae2 at: https://6659d7a000379241510f3765--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 6b1e02a357e4bb8f9c92fdf9f0cba56ca20176f9 at: https://6660354c0a98f0e97f613d9c--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 8b29269c4c4ee7bbc28037b0cb52efce3bf89886 at: https://666395f700616e48b83c524a--meshery-docs-preview.netlify.app

File: docs/pages/guides/infrastructure-management/infrastructure-management.md [hunk 2]
```diff
@@ -43,18 +43,28 @@ See [Importing Applications]({{site.baseurl}}guides/configuration-management/imp
...
 <h4>5. Provisioning</h4>
 <p>A Directed Acyclic Graph (DAG) generated in the previous step is processed. Dependent components are processed sequentially, while others are processed in parallel. Meshery intelligently handles the deployment order to ensure successful deployment.</p>
 </details>
 
 <h3>Auto-Deployment of CRDs and Operators</h3>
+Meshery automates the deployment of Custom Resource Definitions (CRDs) and operators based on the source from which a particular component was registered. When a deployment request is made for a Helm Chart without the upgrade flag, Meshery does not check for the existence of the Helm Chart and always tries to install it. In cases where the chart/release already exists, the installation fails (from the Helm side), and Meshery errors out. Currently, by default, Meshery automatically deploys components that are sourced from Artifact Hub (utilizing Helm Charts).
```
Comment (MUzairS15) [inline docs/pages/guides/infrastructure-management/infrastructure-management.md:52]: Deployment request made for Helm chart?
It happens only when users deploy helm chart (No)
Make it clear, it is confusing.

File: docs/pages/guides/infrastructure-management/infrastructure-management.md [hunk 3]
```diff
@@ -43,18 +43,28 @@ See [Importing Applications]({{site.baseurl}}guides/configuration-management/imp
...
 
-Meshery automates the deployment of Custom Resource Definitions (CRDs) and operators based on the source from which a particular component was registered. Currently, by default Meshery automatically deploys components that are sourced from Artifact Hub (utilizing Helm Charts). (_Support for OCI registries is expected in the near future._)
-
+Meshery gives users control over installing dependencies through a simple checkbox option. Users can decide whether to install dependencies during deployment. By default, Meshery tries to install dependencies without checking for an existing release, making deployment quicker. (Support for OCI registries is coming soon.)
```
Comment (MUzairS15) [inline docs/pages/guides/infrastructure-management/infrastructure-management.md:54]:
```suggestion (Support for OCI registries is coming soon.)
```

what is the relevance of this statement here? You updated the paragraph so do it in entirety.

Where we talk about  helm chart registries add there.

File: docs/pages/guides/infrastructure-management/infrastructure-management.md [hunk 4]
```diff
@@ -43,18 +43,28 @@ See [Importing Applications]({{site.baseurl}}guides/configuration-management/imp
...
   Meshery will soon be adding support for OCI registries in addition to ArtifactHub.
   </p>" %}
+
+<h3>Frequently Asked Questions</h3>
+
+**Q: How does Meshery determine where to get the required dependency for provisioning components?**
+
+**A:** Meshery determines the source based on the metadata of models and components. This metadata includes the actual location of the components, whether it's a Helm chart, Kubernetes manifests, or any other format. Depending on the type of source (Helm chart or Kubernetes manifest), Meshery deploys the components accordingly.
+
+**Q: How does Meshery know what Operator to provision when provisioning components from a given Model?**
```
Comment (MUzairS15) [inline docs/pages/guides/infrastructure-management/infrastructure-management.md:68]: There isn’t a difference on how Meshery determines dependent CRDs and operators.

**Review by MUzairS15 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**